src/lib/snarky_js_bindings/lib/snarky_js_bindings_lib.ml

```diff
@@ -0,0 +1,2673 @@
+    ; totalCurrency : js_uint64 closed_interval Js.prop >
+    Js.t
+
+  type epoch_data_predicate =
+    < ledger : epoch_ledger_predicate Js.prop
+    ; seed : js_field or_ignore Js.prop
+    ; startCheckpoint : js_field or_ignore Js.prop
+    ; lockCheckpoint : js_field or_ignore Js.prop
+    ; epochLength : js_uint32 closed_interval Js.prop >
+    Js.t
+
+  type protocol_state_predicate =
+    < snarkedLedgerHash : js_field or_ignore Js.prop
+    ; timestamp : js_uint64 closed_interval Js.prop
+    ; blockchainLength : js_uint32 closed_interval Js.prop
+    ; minWindowDensity : js_uint32 closed_interval Js.prop
+    ; lastVrfOutput : js_field or_ignore Js.prop
+    ; totalCurrency : js_uint64 closed_interval Js.prop
+    ; globalSlotSinceHardFork : js_uint32 closed_interval Js.prop
+    ; globalSlotSinceGenesis : js_uint32 closed_interval Js.prop
+    ; stakingEpochData : epoch_data_predicate Js.prop
+    ; nextEpochData : epoch_data_predicate Js.prop >
+    Js.t
+
+  type public_key = < g : group_class Js.t Js.prop > Js.t
+
+  type party_update =
+    < appState : js_field set_or_keep Js.js_array Js.t Js.prop
+    ; delegate : public_key set_or_keep Js.prop
+    ; verificationKey : Js.js_string Js.t set_or_keep Js.prop
+    ; votingFor : js_field set_or_keep Js.prop >
+    Js.t
+
+  type js_int64 = < uint64Value : js_field Js.meth > Js.t
+
+  type timing_info =
+    < initialMinimumBalance : js_uint64 Js.prop
+    ; cliffTime : js_uint32 Js.prop
+    ; cliffAmount : js_uint64 Js.prop
+    ; vestingPeriod : js_uint32 Js.prop
+    ; vestingIncrement : js_uint64 Js.prop >
+    Js.t
+
+  type party_body =
+    < publicKey : public_key Js.prop
+    ; update : party_update Js.prop
+    ; tokenId : js_field Js.prop
+    ; delta : js_int64 Js.prop
+    ; events : js_field Js.js_array Js.t Js.js_array Js.t Js.prop
+    ; sequenceEvents : js_field Js.js_array Js.t Js.js_array Js.t Js.prop
+    ; callData : js_field Js.prop
+    ; depth : int Js.prop
+    ; protocolState : protocol_state_predicate Js.prop
+    ; useFullCommitment : js_bool Js.prop
+    ; incrementNonce : js_bool Js.prop >
+    Js.t
+
+  type full_account_predicate =
+    < balance : js_uint64 closed_interval Js.prop
+    ; nonce : js_uint32 closed_interval Js.prop
+    ; receiptChainHash : js_field or_ignore Js.prop
+    ; publicKey : public_key or_ignore Js.prop
+    ; delegate : public_key or_ignore Js.prop
+    ; state : js_field or_ignore Js.js_array Js.t Js.prop
+    ; sequenceState : js_field or_ignore Js.prop
+    ; provedState : bool_class Js.t or_ignore Js.prop >
+    Js.t
+
+  module Party_predicate = struct
+    type party_predicate
+
+    type t =
+      < type_ : Js.js_string Js.t Js.prop ; value : party_predicate Js.prop >
+      Js.t
+  end
+
+  type party =
+    < body : party_body Js.prop ; predicate : Party_predicate.t Js.prop > Js.t
+
+  type fee_payer_party =
+    < body : party_body Js.prop ; predicate : js_uint32 Js.prop > Js.t
+
+  type parties =
```

Comment:
oof, we should definitely autogenerate all these with the derivers system. I know you were already planning this; can you make an issue for this and link to it in a comment?

Reply:
Created issue: https://github.com/MinaProtocol/mina/issues/10553